### PR TITLE
Update renovate.json

### PR DESCRIPTION
**Pull request by Rosie-Brigham**
# Context
[Jira ticket - CAS-509](https://dsdmoj.atlassian.net/browse/CAS-509)

Following Andrew Lee's analysis of Circle CI usage across hmpps, it transpired that we are using a deprecated renovate config. 
This updates the config to that used by the typescript template

### CODE REVIEW

**Comment by aliuk2012** [comment]
@Rosie-Brigham any particular reason why we use renovate instead of say dependabot?

**Review by froddd — DISMISSED** [review]
LGTM! Glad we can get rid of those alerts to move up major versions 🎉

File: renovate.json [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+      "matchManagers": ["npm"],
+      "matchUpdateTypes": ["minor", "patch"],
+      "groupName": "all non major NPM dependencies",
+      "groupSlug": "all-npm-minor-patch",
+      "stabilityDays": 5
+    },
+    {
+      "matchDepTypes": ["engines"],
+      "enabled": false
+    },
```
Comment (froddd) [inline renovate.json]: This seems covered already in the base node config from hmpps: https://github.com/ministryofjustice/hmpps-renovate-config/blob/main/node.json#L6-L9

File: renovate.json [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+      "matchPackageNames": ["typescript", "govuk-frontend"],
+      "rangeStrategy": "bump",
+      "stabilityDays": 0
+    },
+    {
+      "matchManagers": ["npm"],
+      "matchPackageNames": ["@types/node"],
+      "matchUpdateTypes": ["major"],
+      "enabled": false
+    },
```
Comment (froddd) [inline renovate.json:29]: Is this needed given the above `["minor", "patch"]` setting for all `npm` packages?

File: renovate.json [hunk 3]
```diff
@@ -1,3 +1,42 @@
...
+    {
+      "matchDepTypes": ["engines"],
+      "enabled": false
+    },
+    {
+      "matchPackageNames": ["typescript", "govuk-frontend"],
+      "rangeStrategy": "bump",
+      "stabilityDays": 0
+    },
```
Comment (froddd) [inline renovate.json]: If this is an attempt to replicate the functionality from the old `check_outdated` CI script, should this specify the update type and force the check? I might be reading it wrong, but it seems like this on its own won't do anything?

```suggestion
    {
      "matchPackageNames": ["typescript", "govuk-frontend"],
      "matchUpdateTypes": ["major"],
      "enabled": true,
      "rangeStrategy": "bump",
      "stabilityDays": 0
    },
```

File: renovate.json (hunk 2 above)
Comment (Rosie-Brigham) [reply renovate.json:29]: In all honesty - I am unsure, I was essentially just told to copy this into the config. I have never come across rennovate before so still learning the config across the dept

**Review by froddd — DISMISSED** [review]
Looks good!

File: doc/architecture/decisions/0010-update-rennovate-congif.md
Comment (froddd) [inline doc/architecture/decisions/0010-update-rennovate-congif.md:1]: Is this meant to include the actual decision on the update strategy?

Comment (Rosie-Brigham) [reply doc/architecture/decisions/0010-update-rennovate-congif.md:1]: 🤦‍♀️

File: doc/architecture/decisions/0010-update-rennovate-congif.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# 8. concept-of-bedspaces
```
Comment (froddd) [inline doc/architecture/decisions/0010-update-rennovate-congif.md]: Should this be:

```suggestion
# 10. Update Renovate config
```

**Review by froddd — DISMISSED** [review]
One update to ADR title needed, otherwise good to go!

**Review by froddd — APPROVED** [review]
👍

**Pull request merged**